row selection + some styling

kaperoo opened the pull request:
## Description
Add row selection to the table.

Since we use server-side pagination, it is not possible to select all rows with TanStack (known [issue](https://github.com/TanStack/table/issues/4781)).
For now the checkbox in the header of the table toggles selection for all rows on the current page. 

Also, added a bit of styling to the table cells so that everything is aligned:
![image](https://github.com/ral-facilities/operationsgateway/assets/72079517/59492897-d2b0-4e86-845b-cfdfeffa9d02)

Temporarily (?) switched off the background colour of the columns so that the highlighting of selected rows is visible.

## Testing instructions
Selecting rows on one page shouldn't select (or unselect) any rows on any other page.

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
connect to DSEGOG-120

kaperoo committed:
row selection + some styling

kaperoo committed:
update unit tests + snapshots

kaperoo committed:
Merge branch 'develop' into DSEGOG-120-export-row-selection

kaperoo committed:
fix unit test console error

kaperoo committed:
missed 2 cases

kaperoo committed:
fix cypress e2e tests

kaperoo committed:
updated snapshots

kaperoo committed:
Merge branch 'develop' into DSEGOG-120-export-row-selection

kaperoo committed:
row selection tests

louise-davies commented on src/table/table.component.tsx:
```diff
@@ -154,6 +163,22 @@ const Table = React.memo((props: TableProps): React.ReactElement => {
                         >
+                          <MuiTableCell
+                            padding="checkbox"
+                            sx={{ width: '48px' }}
```
Removing these manually set widths doesn't affect the styling at all as far as I can see, so we should probably remove them

louise-davies commented on src/table/table.component.tsx:
```diff
@@ -215,16 +240,31 @@ const Table = React.memo((props: TableProps): React.ReactElement => {
                 >
+                  <MuiTableCell
+                    padding="checkbox"
+                    sx={{ width: '48px' }}
```
See above

louise-davies commented on src/table/table.component.tsx:
```diff
@@ -215,16 +240,31 @@ const Table = React.memo((props: TableProps): React.ReactElement => {
+                    sx={{ width: '48px' }}
+                    scope="row"
+                  >
+                    <Checkbox
```
These checkboxes look a little big/have too much padding imo - especially in a text only table. Can we experiment with size small checkboxes and reducing their padding (which reduced the size of the ripple effect)

Also, they aren't vertically aligned when an image or trace channel is present

kaperoo commented:
Changed checkbox size to _small_ and reduced their padding:
![image](https://github.com/ral-facilities/operationsgateway/assets/72079517/79bbe65e-edc7-4c08-9c1b-4fb2e4c197b8)

Also, checkboxes are now vertically aligned:
![image](https://github.com/ral-facilities/operationsgateway/assets/72079517/c01f5518-50be-482f-aa05-5944f188e3bf)

kaperoo committed:
changed checkbox sizes and padding

kaperoo committed:
Merge branch 'develop' into DSEGOG-120-export-row-selection

kaperoo committed:
updated snapshots

louise-davies commented on src/table/table.component.tsx:
```diff
@@ -239,16 +264,36 @@ const Table = React.memo((props: TableProps): React.ReactElement => {
+                    <Checkbox
+                      size="small"
+                      inputProps={{ 'aria-label': 'select row' }}
+                      sx={{ paddingTop: '5px', paddingBottom: '5px' }}
```
Minor nitpick, but this messes with the ripple shape (see below)

![image](https://github.com/ral-facilities/operationsgateway/assets/22525471/f9c7087e-634e-47bc-b85c-15e07c71bc37)

I believe if we set padding on all sides to 5, and then add back in the extra space on the left and right with margin 4px it fixes it e.g.
`sx={{ p: "5px", m: "0 4px" }}`

![image](https://github.com/ral-facilities/operationsgateway/assets/22525471/27056453-08ad-4786-a068-06bac65b9890)

kaperoo replied:
Yeah, I see it now. I'll also add padding + margins to the _"select all"_ checkbox so that the ripple sizes are the same.

kaperoo committed:
fixed padding + margins

kaperoo committed:
updated snapshots

louise-davies committed:
DSEGOG-120 integrate sticky columns with checkbox column
Make react table aware of the checkbox column, so we can work out sticky styles properly, i.e. the selection/hover styles
and making the checkbox column itself also sticky
For some reason we have to use column pinning for it to work - idk why
Also, fixed a width issue where we weren't taking the sidebar into account
This also has a side effect of nicely grouping the header and cell checkbox logic together

louise-davies reviewed (approved):
Kacper has tested my code and both the sticky columns & the row selection seem to be working well with all the other table features now
